File: 0229-majority-element-ii/0229-majority-element-ii.rb
# @param {Integer[]} nums
# @return {Integer[]}
def majority_element(nums)
    dict = Hash.new(0)
    nums.each do |i|
      dict[i] += 1
    end

    res = []
    compare = nums.length / 3
    dict.each do |key, value|
      res << key if value > compare
    end

    res
  end

File: 0229-majority-element-ii/0229-majority-element-ii.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function majorityElement($nums) {
        $dict = [];
        foreach ($nums as $i) {
            if (isset($dict[$i])) {
                $dict[$i]++;
            } else {
                $dict[$i] = 1;
            }
        }

        $res = [];
        $compare = count($nums) / 3;
        foreach ($dict as $key => $value) {
            if ($value > $compare) {
                $res[] = $key;
            }
        }

        return $res;
    }
    }